### PR TITLE
Allow more uses of `beforeInteractive` strategy in Script component

### LINKED ISSUE #31275: Unable to use `dangerouslySetInnerHTML` and `beforeInteractive` strategy in Script component
### What version of Next.js are you using?

14.17.1

### What version of Node.js are you using?

12.0.3

### What browser are you using?

Chrome

### What operating system are you using?

macOS

### How are you deploying your application?

npm run dev

### Describe the Bug

Setting `dangerouslySetInnerHtml` in next Script component while also specifying `beforeInteractive` strategy will result in the Script component doing nothing. Since there is no `src` specified, the src will simply be `(unknown)`.
script

```
<Script
  id="test"
  strategy="beforeInteractive"
  dangerouslySetInnerHTML={{
    // will not run
    __html: `console.log('hello!')`,
  }}
/>
```

### Expected Behavior

Script component can be used with no `src` while using `dangerouslySetInnerHtml` property. 

### To Reproduce

Try to render Script component using `beforeInteractive` while having no `url` and specifying `dangerouslySetInnerHtml` property.

```
<Script
  id="test"
  strategy="beforeInteractive"
  dangerouslySetInnerHTML={{
    // will not run
    __html: `console.log('hello!')`,
  }}
/>
```
This piece of code of next Script will result in the following script tag being added
script

### PR DESCRIPTION
**Pull request by genetschneider**
This PR ensures Script component can be used in `beforeInteractive` strategy even if `src` is not specified.
- Fixes [#31275](https://github.com/vercel/next.js/issues/31275)

### CODE REVIEW

File: test/integration/script-loader/test/index.test.js [hunk 1]
```diff
@@ -118,9 +108,16 @@ describe('Script Loader', () => {
       expect(
         $(`#${id} ~ script[src^="/_next/static/chunks/main"]`).length
       ).toBeGreaterThan(0)
+
+      if (id === 'scriptBeforeInteractiveDangerous') {
+        expect(script.contents().first().text()).toEqual('//hello')
+        expect(script.attr('src')).toEqual(undefined)
+        expect(typeof script.attr('src')).toBe('undefined')
```
Comment (genetschneider) [inline test/integration/script-loader/test/index.test.js:115]: ensure the type is undefined not an empty string

**Comment by ijjk** [comment]
# Stats from current PR

<details>
<summary><strong>Default Build</strong> (Decrease detected ✓)</summary>

<br/>

<details>
<summary><strong>General</strong> Overall increase ⚠️</summary>

|  | vercel/next.js canary  | genetschneider/next.js-1 fix-before-interactive-script | Change |
| - | - | - | - |
| buildDuration | 19.7s | 20.2s | ⚠️ +464ms |
| buildDurationCached | 4s | 4.1s | ⚠️ +91ms |
| nodeModulesSize | 334 MB | 334 MB | ⚠️ +281 B |

</details>

<details>
<summary><strong>Page Load Tests</strong> Overall decrease ⚠️</summary>

|  | vercel/next.js canary  | genetschneider/next.js-1 fix-before-interactive-script | Change |
| - | - | - | - |
| / failed reqs | 0 | 0 | ✓ |
| / total time (seconds) | 3.253 | 3.286 | ⚠️ +0.03 |
| / avg req/sec | 768.62 | 760.75 | ⚠️ -7.87 |
| /error-in-render failed reqs | 0 | 0 | ✓ |
| /error-in-render total time (seconds) | 1.512 | 1.571 | ⚠️ +0.06 |
| /error-in-render avg req/sec | 1653.95 | 1591.01 | ⚠️ -62.94 |

</details>

<details>
<summary><strong>Client Bundles (main, webpack, commons)</strong> Overall decrease ✓</summary>

|  | vercel/next.js canary  | genetschneider/next.js-1 fix-before-interactive-script | Change |
| - | - | - | - |
| 450.HASH.js gzip | 179 B | 179 B | ✓ |
| framework-HASH.js gzip | 42.2 kB | 42.2 kB | ✓ |
| main-HASH.js gzip | 28.2 kB | 28.2 kB | -3 B |
| webpack-HASH.js gzip | 1.45 kB | 1.45 kB | ✓ |
| Overall change | 72.1 kB | 72.1 kB | -3 B |

</details>

<details>
<summary><strong>Legacy Client Bundles (polyfills)</strong></summary>

|  | vercel/next.js canary  | genetschneider/next.js-1 fix-before-interactive-script | Change |
| - | - | - | - |
| polyfills-HASH.js gzip | 31 kB | 31 kB | ✓ |
| Overall change | 31 kB | 31 kB | ✓ |

</details>

<details>
<summary><strong>Client Pages</strong></summary>

|  | vercel/next.js canary  | genetschneider/next.js-1 fix-before-interactive-script | Change |
| - | - | - | - |
| _app-HASH.js gzip | 1.37 kB | 1.37 kB | ✓ |
| _error-HASH.js gzip | 194 B | 194 B | ✓ |
| amp-HASH.js gzip | 312 B | 312 B | ✓ |
| css-HASH.js gzip | 327 B | 327 B | ✓ |
| dynamic-HASH.js gzip | 2.38 kB | 2.38 kB | ✓ |
| head-HASH.js gzip | 350 B | 350 B | ✓ |
| hooks-HASH.js gzip | 635 B | 635 B | ✓ |
| image-HASH.js gzip | 4.44 kB | 4.44 kB | ✓ |
| index-HASH.js gzip | 263 B | 263 B | ✓ |
| link-HASH.js gzip | 1.87 kB | 1.87 kB | ✓ |
| routerDirect..HASH.js gzip | 321 B | 321 B | ✓ |
| script-HASH.js gzip | 383 B | 383 B | ✓ |
| withRouter-HASH.js gzip | 318 B | 318 B | ✓ |
| 85e02e95b279..7e3.css gzip | 107 B | 107 B | ✓ |
| Overall change | 13.3 kB | 13.3 kB | ✓ |

</details>

<details>
<summary><strong>Client Build Manifests</strong></summary>

|  | vercel/next.js canary  | genetschneider/next.js-1 fix-before-interactive-script | Change |
| - | - | - | - |
| _buildManifest.js gzip | 460 B | 460 B | ✓ |
| Overall change | 460 B | 460 B | ✓ |

</details>

<details>
<summary><strong>Rendered Page Sizes</strong> Overall increase ⚠️</summary>

|  | vercel/next.js canary  | genetschneider/next.js-1 fix-before-interactive-script | Change |
| - | - | - | - |
| index.html gzip | 522 B | 524 B | ⚠️ +2 B |
| link.html gzip | 535 B | 536 B | ⚠️ +1 B |
| withRouter.html gzip | 516 B | 517 B | ⚠️ +1 B |
| Overall change | 1.57 kB | 1.58 kB | ⚠️ +4 B |

</details>

#### Diffs
<details>
<summary>Diff for <strong>main-HASH.js</strong></summary>

```diff
@@ -2949,7 +2949,8 @@
                 {
                   src: src,
                   onLoad: onLoad,
-                  onError: onError
+                  onError: onError,
+                  dangerouslySetInnerHTML: dangerouslySetInnerHTML
                 },
                 restProps
               )
```
</details>
<details>
<summary>Diff for <strong>index.html</strong></summary>

```diff
@@ -19,7 +19,7 @@
       defer=""
     ></script>
     <script
-      src="/_next/static/chunks/main-6b0f14463d84c5dc.js"
+      src="/_next/static/chunks/main-31d45e48b2a1672f.js"
       defer=""
     ></script>
     <script
```
</details>
<details>
<summary>Diff for <strong>link.html</strong></summary>

```diff
@@ -19,7 +19,7 @@
       defer=""
     ></script>
     <script
-      src="/_next/static/chunks/main-6b0f14463d84c5dc.js"
+      src="/_next/static/chunks/main-31d45e48b2a1672f.js"
       defer=""
     ></script>
     <script
```
</details>
<details>
<summary>Diff for <strong>withRouter.html</strong></summary>

```diff
@@ -19,7 +19,7 @@
       defer=""
     ></script>
     <script
-      src="/_next/static/chunks/main-6b0f14463d84c5dc.js"
+      src="/_next/static/chunks/main-31d45e48b2a1672f.js"
       defer=""
     ></script>
     <script
```
</details>
</details>
<hr/>
<details>
<summary><strong>Default Build with SWC</strong> (Decrease detected ✓)</summary>

<br/>

<details>
<summary><strong>General</strong> Overall increase ⚠️</summary>

|  | vercel/next.js canary  | genetschneider/next.js-1 fix-before-interactive-script | Change |
| - | - | - | - |
| buildDuration | 21.8s | 21.8s | -6ms |
| buildDurationCached | 4.2s | 4.1s | -161ms |
| nodeModulesSize | 334 MB | 334 MB | ⚠️ +281 B |

</details>

<details>
<summary><strong>Page Load Tests</strong> Overall decrease ⚠️</summary>

|  | vercel/next.js canary  | genetschneider/next.js-1 fix-before-interactive-script | Change |
| - | - | - | - |
| / failed reqs | 0 | 0 | ✓ |
| / total time (seconds) | 3.37 | 3.333 | -0.04 |
| / avg req/sec | 741.91 | 750.07 | +8.16 |
| /error-in-render failed reqs | 0 | 0 | ✓ |
| /error-in-render total time (seconds) | 1.584 | 1.594 | ⚠️ +0.01 |
| /error-in-render avg req/sec | 1578.77 | 1568.38 | ⚠️ -10.39 |

</details>

<details>
<summary><strong>Client Bundles (main, webpack, commons)</strong> Overall increase ⚠️</summary>

|  | vercel/next.js canary  | genetschneider/next.js-1 fix-before-interactive-script | Change |
| - | - | - | - |
| 450.HASH.js gzip | 179 B | 179 B | ✓ |
| framework-HASH.js gzip | 42.3 kB | 42.3 kB | ✓ |
| main-HASH.js gzip | 28.4 kB | 28.4 kB | ⚠️ +5 B |
| webpack-HASH.js gzip | 1.43 kB | 1.43 kB | ✓ |
| Overall change | 72.4 kB | 72.4 kB | ⚠️ +5 B |

</details>

<details>
<summary><strong>Legacy Client Bundles (polyfills)</strong></summary>

|  | vercel/next.js canary  | genetschneider/next.js-1 fix-before-interactive-script | Change |
| - | - | - | - |
| polyfills-HASH.js gzip | 31 kB | 31 kB | ✓ |
| Overall change | 31 kB | 31 kB | ✓ |

</details>

<details>
<summary><strong>Client Pages</strong></summary>

|  | vercel/next.js canary  | genetschneider/next.js-1 fix-before-interactive-script | Change |
| - | - | - | - |
| _app-HASH.js gzip | 1.35 kB | 1.35 kB | ✓ |
| _error-HASH.js gzip | 180 B | 180 B | ✓ |
| amp-HASH.js gzip | 305 B | 305 B | ✓ |
| css-HASH.js gzip | 321 B | 321 B | ✓ |
| dynamic-HASH.js gzip | 2.38 kB | 2.38 kB | ✓ |
| head-HASH.js gzip | 342 B | 342 B | ✓ |
| hooks-HASH.js gzip | 622 B | 622 B | ✓ |
| image-HASH.js gzip | 4.46 kB | 4.46 kB | ✓ |
| index-HASH.js gzip | 256 B | 256 B | ✓ |
| link-HASH.js gzip | 1.91 kB | 1.91 kB | ✓ |
| routerDirect..HASH.js gzip | 314 B | 314 B | ✓ |
| script-HASH.js gzip | 375 B | 375 B | ✓ |
| withRouter-HASH.js gzip | 309 B | 309 B | ✓ |
| 85e02e95b279..7e3.css gzip | 107 B | 107 B | ✓ |
| Overall change | 13.2 kB | 13.2 kB | ✓ |

</details>

<details>
<summary><strong>Client Build Manifests</strong></summary>

|  | vercel/next.js canary  | genetschneider/next.js-1 fix-before-interactive-script | Change |
| - | - | - | - |
| _buildManifest.js gzip | 459 B | 459 B | ✓ |
| Overall change | 459 B | 459 B | ✓ |

</details>

<details>
<summary><strong>Rendered Page Sizes</strong> Overall increase ⚠️</summary>

|  | vercel/next.js canary  | genetschneider/next.js-1 fix-before-interactive-script | Change |
| - | - | - | - |
| index.html gzip | 522 B | 523 B | ⚠️ +1 B |
| link.html gzip | 534 B | 535 B | ⚠️ +1 B |
| withRouter.html gzip | 515 B | 516 B | ⚠️ +1 B |
| Overall change | 1.57 kB | 1.57 kB | ⚠️ +3 B |

</details>

#### Diffs
<details>
<summary>Diff for <strong>main-HASH.js</strong></summary>

```diff
@@ -2949,7 +2949,8 @@
                 {
                   src: src,
                   onLoad: onLoad,
-                  onError: onError
+                  onError: onError,
+                  dangerouslySetInnerHTML: dangerouslySetInnerHTML
                 },
                 restProps
               )
```
</details>
<details>
<summary>Diff for <strong>index.html</strong></summary>

```diff
@@ -19,7 +19,7 @@
       defer=""
     ></script>
     <script
-      src="/_next/static/chunks/main-6b0f14463d84c5dc.js"
+      src="/_next/static/chunks/main-31d45e48b2a1672f.js"
       defer=""
     ></script>
     <script
```
</details>
<details>
<summary>Diff for <strong>link.html</strong></summary>

```diff
@@ -19,7 +19,7 @@
       defer=""
     ></script>
     <script
-      src="/_next/static/chunks/main-6b0f14463d84c5dc.js"
+      src="/_next/static/chunks/main-31d45e48b2a1672f.js"
       defer=""
     ></script>
     <script
```
</details>
<details>
<summary>Diff for <strong>withRouter.html</strong></summary>

```diff
@@ -19,7 +19,7 @@
       defer=""
     ></script>
     <script
-      src="/_next/static/chunks/main-6b0f14463d84c5dc.js"
+      src="/_next/static/chunks/main-31d45e48b2a1672f.js"
       defer=""
     ></script>
     <script
```
</details>
</details>
Commit: aa6cc9814a137479b3c8a5888899363ee20dff49

**Comment by ijjk** [comment]
# Stats from current PR

<details>
<summary><strong>Default Build</strong> (Increase detected ⚠️)</summary>

<br/>

<details>
<summary><strong>General</strong> Overall increase ⚠️</summary>

|  | vercel/next.js canary  | genetschneider/next.js-1 fix-before-interactive-script | Change |
| - | - | - | - |
| buildDuration | 17.1s | 17.3s | ⚠️ +127ms |
| buildDurationCached | 3.3s | 3.4s | ⚠️ +94ms |
| nodeModulesSize | 345 MB | 345 MB | ⚠️ +281 B |

</details>

<details>
<summary><strong>Page Load Tests</strong> Overall increase ✓</summary>

|  | vercel/next.js canary  | genetschneider/next.js-1 fix-before-interactive-script | Change |
| - | - | - | - |
| / failed reqs | 0 | 0 | ✓ |
| / total time (seconds) | 2.743 | 2.756 | ⚠️ +0.01 |
| / avg req/sec | 911.52 | 907.09 | ⚠️ -4.43 |
| /error-in-render failed reqs | 0 | 0 | ✓ |
| /error-in-render total time (seconds) | 1.203 | 1.172 | -0.03 |
| /error-in-render avg req/sec | 2078.71 | 2132.87 | +54.16 |

</details>

<details>
<summary><strong>Client Bundles (main, webpack, commons)</strong></summary>

|  | vercel/next.js canary  | genetschneider/next.js-1 fix-before-interactive-script | Change |
| - | - | - | - |
| 450.HASH.js gzip | 179 B | 179 B | ✓ |
| framework-HASH.js gzip | 42.2 kB | 42.2 kB | ✓ |
| main-HASH.js gzip | 28.4 kB | 28.4 kB | ✓ |
| webpack-HASH.js gzip | 1.45 kB | 1.45 kB | ✓ |
| Overall change | 72.2 kB | 72.2 kB | ✓ |

</details>

<details>
<summary><strong>Legacy Client Bundles (polyfills)</strong></summary>

|  | vercel/next.js canary  | genetschneider/next.js-1 fix-before-interactive-script | Change |
| - | - | - | - |
| polyfills-HASH.js gzip | 31 kB | 31 kB | ✓ |
| Overall change | 31 kB | 31 kB | ✓ |

</details>

<details>
<summary><strong>Client Pages</strong></summary>

|  | vercel/next.js canary  | genetschneider/next.js-1 fix-before-interactive-script | Change |
| - | - | - | - |
| _app-HASH.js gzip | 1.37 kB | 1.37 kB | ✓ |
| _error-HASH.js gzip | 194 B | 194 B | ✓ |
| amp-HASH.js gzip | 312 B | 312 B | ✓ |
| css-HASH.js gzip | 327 B | 327 B | ✓ |
| dynamic-HASH.js gzip | 2.38 kB | 2.38 kB | ✓ |
| head-HASH.js gzip | 350 B | 350 B | ✓ |
| hooks-HASH.js gzip | 635 B | 635 B | ✓ |
| image-HASH.js gzip | 4.45 kB | 4.45 kB | ✓ |
| index-HASH.js gzip | 263 B | 263 B | ✓ |
| link-HASH.js gzip | 1.87 kB | 1.87 kB | ✓ |
| routerDirect..HASH.js gzip | 321 B | 321 B | ✓ |
| script-HASH.js gzip | 383 B | 383 B | ✓ |
| withRouter-HASH.js gzip | 318 B | 318 B | ✓ |
| 85e02e95b279..7e3.css gzip | 107 B | 107 B | ✓ |
| Overall change | 13.3 kB | 13.3 kB | ✓ |

</details>

<details>
<summary><strong>Client Build Manifests</strong></summary>

|  | vercel/next.js canary  | genetschneider/next.js-1 fix-before-interactive-script | Change |
| - | - | - | - |
| _buildManifest.js gzip | 460 B | 460 B | ✓ |
| Overall change | 460 B | 460 B | ✓ |

</details>

<details>
<summary><strong>Rendered Page Sizes</strong> Overall increase ⚠️</summary>

|  | vercel/next.js canary  | genetschneider/next.js-1 fix-before-interactive-script | Change |
| - | - | - | - |
| index.html gzip | 531 B | 532 B | ⚠️ +1 B |
| link.html gzip | 545 B | 545 B | ✓ |
| withRouter.html gzip | 525 B | 526 B | ⚠️ +1 B |
| Overall change | 1.6 kB | 1.6 kB | ⚠️ +2 B |

</details>

#### Diffs
<details>
<summary>Diff for <strong>main-HASH.js</strong></summary>

```diff
@@ -2984,7 +2984,8 @@
                 {
                   src: src,
                   onLoad: onLoad,
-                  onError: onError
+                  onError: onError,
+                  dangerouslySetInnerHTML: dangerouslySetInnerHTML
                 },
                 restProps
               )
```
</details>
<details>
<summary>Diff for <strong>index.html</strong></summary>

```diff
@@ -19,7 +19,7 @@
       defer=""
     ></script>
     <script
-      src="/_next/static/chunks/main-966353a8b06621ca.js"
+      src="/_next/static/chunks/main-6723a452c6f77e47.js"
       defer=""
     ></script>
     <script
```
</details>
<details>
<summary>Diff for <strong>link.html</strong></summary>

```diff
@@ -19,7 +19,7 @@
       defer=""
     ></script>
     <script
-      src="/_next/static/chunks/main-966353a8b06621ca.js"
+      src="/_next/static/chunks/main-6723a452c6f77e47.js"
       defer=""
     ></script>
     <script
```
</details>
<details>
<summary>Diff for <strong>withRouter.html</strong></summary>

```diff
@@ -19,7 +19,7 @@
       defer=""
     ></script>
     <script
-      src="/_next/static/chunks/main-966353a8b06621ca.js"
+      src="/_next/static/chunks/main-6723a452c6f77e47.js"
       defer=""
     ></script>
     <script
```
</details>
</details>
<hr/>
<details>
<summary><strong>Default Build with SWC</strong> (Decrease detected ✓)</summary>

<br/>

<details>
<summary><strong>General</strong> Overall increase ⚠️</summary>

|  | vercel/next.js canary  | genetschneider/next.js-1 fix-before-interactive-script | Change |
| - | - | - | - |
| buildDuration | 18.8s | 19.1s | ⚠️ +288ms |
| buildDurationCached | 3.4s | 3.4s | ⚠️ +24ms |
| nodeModulesSize | 345 MB | 345 MB | ⚠️ +281 B |

</details>

<details>
<summary><strong>Page Load Tests</strong> Overall decrease ⚠️</summary>

|  | vercel/next.js canary  | genetschneider/next.js-1 fix-before-interactive-script | Change |
| - | - | - | - |
| / failed reqs | 0 | 0 | ✓ |
| / total time (seconds) | 2.741 | 2.777 | ⚠️ +0.04 |
| / avg req/sec | 912.13 | 900.13 | ⚠️ -12 |
| /error-in-render failed reqs | 0 | 0 | ✓ |
| /error-in-render total time (seconds) | 1.167 | 1.217 | ⚠️ +0.05 |
| /error-in-render avg req/sec | 2141.87 | 2053.58 | ⚠️ -88.29 |

</details>

<details>
<summary><strong>Client Bundles (main, webpack, commons)</strong> Overall increase ⚠️</summary>

|  | vercel/next.js canary  | genetschneider/next.js-1 fix-before-interactive-script | Change |
| - | - | - | - |
| 450.HASH.js gzip | 179 B | 179 B | ✓ |
| framework-HASH.js gzip | 42.3 kB | 42.3 kB | ✓ |
| main-HASH.js gzip | 28.6 kB | 28.6 kB | ⚠️ +5 B |
| webpack-HASH.js gzip | 1.44 kB | 1.44 kB | ✓ |
| Overall change | 72.5 kB | 72.5 kB | ⚠️ +5 B |

</details>

<details>
<summary><strong>Legacy Client Bundles (polyfills)</strong></summary>

|  | vercel/next.js canary  | genetschneider/next.js-1 fix-before-interactive-script | Change |
| - | - | - | - |
| polyfills-HASH.js gzip | 31 kB | 31 kB | ✓ |
| Overall change | 31 kB | 31 kB | ✓ |

</details>

<details>
<summary><strong>Client Pages</strong></summary>

|  | vercel/next.js canary  | genetschneider/next.js-1 fix-before-interactive-script | Change |
| - | - | - | - |
| _app-HASH.js gzip | 1.35 kB | 1.35 kB | ✓ |
| _error-HASH.js gzip | 180 B | 180 B | ✓ |
| amp-HASH.js gzip | 305 B | 305 B | ✓ |
| css-HASH.js gzip | 321 B | 321 B | ✓ |
| dynamic-HASH.js gzip | 2.37 kB | 2.37 kB | ✓ |
| head-HASH.js gzip | 342 B | 342 B | ✓ |
| hooks-HASH.js gzip | 622 B | 622 B | ✓ |
| image-HASH.js gzip | 4.47 kB | 4.47 kB | ✓ |
| index-HASH.js gzip | 256 B | 256 B | ✓ |
| link-HASH.js gzip | 1.91 kB | 1.91 kB | ✓ |
| routerDirect..HASH.js gzip | 314 B | 314 B | ✓ |
| script-HASH.js gzip | 375 B | 375 B | ✓ |
| withRouter-HASH.js gzip | 309 B | 309 B | ✓ |
| 85e02e95b279..7e3.css gzip | 107 B | 107 B | ✓ |
| Overall change | 13.2 kB | 13.2 kB | ✓ |

</details>

<details>
<summary><strong>Client Build Manifests</strong></summary>

|  | vercel/next.js canary  | genetschneider/next.js-1 fix-before-interactive-script | Change |
| - | - | - | - |
| _buildManifest.js gzip | 458 B | 458 B | ✓ |
| Overall change | 458 B | 458 B | ✓ |

</details>

<details>
<summary><strong>Rendered Page Sizes</strong> Overall decrease ✓</summary>

|  | vercel/next.js canary  | genetschneider/next.js-1 fix-before-interactive-script | Change |
| - | - | - | - |
| index.html gzip | 533 B | 532 B | -1 B |
| link.html gzip | 546 B | 545 B | -1 B |
| withRouter.html gzip | 527 B | 527 B | ✓ |
| Overall change | 1.61 kB | 1.6 kB | -2 B |

</details>

#### Diffs
<details>
<summary>Diff for <strong>main-HASH.js</strong></summary>

```diff
@@ -2984,7 +2984,8 @@
                 {
                   src: src,
                   onLoad: onLoad,
-                  onError: onError
+                  onError: onError,
+                  dangerouslySetInnerHTML: dangerouslySetInnerHTML
                 },
                 restProps
               )
```
</details>
<details>
<summary>Diff for <strong>index.html</strong></summary>

```diff
@@ -19,7 +19,7 @@
       defer=""
     ></script>
     <script
-      src="/_next/static/chunks/main-966353a8b06621ca.js"
+      src="/_next/static/chunks/main-6723a452c6f77e47.js"
       defer=""
     ></script>
     <script
```
</details>
<details>
<summary>Diff for <strong>link.html</strong></summary>

```diff
@@ -19,7 +19,7 @@
       defer=""
     ></script>
     <script
-      src="/_next/static/chunks/main-966353a8b06621ca.js"
+      src="/_next/static/chunks/main-6723a452c6f77e47.js"
       defer=""
     ></script>
     <script
```
</details>
<details>
<summary>Diff for <strong>withRouter.html</strong></summary>

```diff
@@ -19,7 +19,7 @@
       defer=""
     ></script>
     <script
-      src="/_next/static/chunks/main-966353a8b06621ca.js"
+      src="/_next/static/chunks/main-6723a452c6f77e47.js"
       defer=""
     ></script>
     <script
```
</details>
</details>
Commit: 0d09cad07797c45c151ec9ed6ad3ede01ece86e6

**Comment by janicklas-ralph** [comment]
@genetschneider Thanks for this PR. 
I am changing the implementation of beforeInteractive strategy to make it work for the next versions of react - Check this [PR](https://github.com/vercel/next.js/pull/28889)

I'm closing this Pr out since it won't be compatible with the new change.

**Pull request closed without merging**